### PR TITLE
Mrc

### PR DESCRIPTION
**Pull request by mpks**
Hi, 

This PR should be an MRC to Nexus file converter that I added to nexgen. It is just a single command (and a single script) that I wrote to be similar to `ED_nexus`. The basic usage would be  `ED_mrc_to_nexus *.mrc` where `*.mrc` expands in the shell to all mrc files in the current directory. There are other optional arguments to overwrite starting angle, beam center etc. I've tested this on a list of like 100 mrc files and my dials output from nexgen is the same as with MRC files.

The current version of the converter takes a list of MRC images and converts them into a data h5 file with LZ4 compression. It then creates a master similar to the one for the Singla detector.

### CODE REVIEW

File: src/nexgen/nxs_utils/Detector.py [hunk 1]
```diff
@@ -137,6 +145,30 @@ def hasMeta(self) -> bool:
...
+class CetaDetector(DataClassJsonMixin):
+    """Define a Ceta-D detector."""
+
+    description: str
+    image_size: List[float] | Tuple[float]
+    pixel_size: List[str | float] = field(
+        default_factory=lambda: ["0.014mm", "0.014mm"]
+    )
+    sensor_material: str = "Si"
+    sensor_thickness: str = "0.00000000000001mm"
```
Comment (noemifrisina) [inline src/nexgen/nxs_utils/Detector.py:158]: 10^-14 mm doesn't sound right for a sensor thickness. Docs seem to point at something around 14um, could it be something similar?

File: src/nexgen/command_line/ED_mrc_to_nexus.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    gonio_axes = EDCeta.gonio
+    gonio_axes[0].start_pos = start_angle
+    gonio_axes[0].increment = increment
+    gonio_axes[0].num_steps = tot_imgs
+
+    scan = calculate_scan_points(gonio_axes[0],
+                                 rotation=True,
+                                 tot_num_imgs=tot_imgs)
+    goniometer = Goniometer(gonio_axes, scan)
+    osc, trans = goniometer.define_scan_from_goniometer_axes()
```
Comment (noemifrisina) [inline src/nexgen/command_line/ED_mrc_to_nexus.py:111]: Sorry, forgot a comment.

Must: remove this as it is not needed. When a scan is passed, the Goniometer is automatically updated. This particular function instead is only needed when you don't already have a scan and want the Goniometer to calculate it for you from the start_pos/increment/num_steps values of the axes.

File: src/nexgen/nxs_utils/Detector.py (hunk 1 above)
Comment (mpks) [reply src/nexgen/nxs_utils/Detector.py:158]: Yes, I fixed this. This is probably not used anywhere so it won't cause any problems. I think some parameter 'mu' is computed from this, but it is not used.

File: src/nexgen/command_line/ED_mrc_to_nexus.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+    detector = Detector(det_params, det_axes, beam_center,
+                        exposure_time,
+                        [EDCeta.fast_axis,
+                         EDCeta.slow_axis])
+
+    source = EDSource
+    source.beamline = 'Ceta'
+
+    script_dir = os.getcwd()
```
[the remark this answers is not in the document]
Comment (mpks) [reply src/nexgen/command_line/ED_mrc_to_nexus.py]: Yes. I fixed this now. The Nexus file is now saved in the same directory as MRC images.

File: src/nexgen/command_line/ED_mrc_to_nexus.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+         tot_imgs,
+         ED_coord_system)
+
+    datafiles = [data_path]
+    writer.write(datafiles, '/entry/data/data')
+    writer.write_vds(vds_dtype=np.int32,
+                     datafiles=datafiles)
+
+
+def get_metadata(mrc_image):
```
[the remark this answers is not in the document]
Comment (mpks) [reply src/nexgen/command_line/ED_mrc_to_nexus.py]: Done.

**Review by noemifrisina — APPROVED** [review]
Looks okay. As mentioned it would be tidier for this CLI and the singla one to be merged but will open a new ticket for it

**Pull request merged**